state/factory/workingset_test.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/iotexproject/iotex-core/test/identityset"
+	"github.com/iotexproject/iotex-core/testutil"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+	"time"
+	//"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	//"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+func TestWriteReadView(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	ctx := protocol.WithBlockchainCtx(
+		protocol.WithRegistry(context.Background(), protocol.NewRegistry()),
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(ctx))
+	ws, err := sf.(workingSetCreator).newWorkingSet(ctx, 1)
+
+	t.Run("ReadView call error", func(t *testing.T) {
+		retrdf, retErr := ws.ReadView("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
+		require.Equal(retrdf, nil)
+	})
+
+	t.Run("WriteView ReadView normal call ", func(t *testing.T) {
+		require.NoError(ws.WriteView("test", "test"))
+		retrdf, retErr := ws.ReadView("test")
+		require.NoError(retErr)
+		require.Equal(retrdf, "test")
+	})
+
+	t.Run("Unload call error", func(t *testing.T) {
+		retrdf, retErr := ws.Unload("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
+		require.Equal(retrdf, nil)
+	})
+
+	t.Run("Load Unload normal call ", func(t *testing.T) {
+		require.NoError(ws.Load("test", "test"))
+		retrdf, retErr := ws.Unload("test")
+		require.NoError(retErr)
+		require.Equal(retrdf, "test")
+	})
+}
+
+func TestProtocolDirty(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	ctx := protocol.WithBlockchainCtx(
+		protocol.WithRegistry(context.Background(), protocol.NewRegistry()),
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(ctx))
+	ws, err := sf.(workingSetCreator).newWorkingSet(ctx, 1)
+	require.False(ws.ProtocolDirty("test"))
+
+	require.NoError(ws.Load("test", "test"))
+	require.True(ws.ProtocolDirty("test"))
+}
+
+func TestValidateBlock(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	gasLimit := testutil.TestGasLimit * 100000
+	zctx := protocol.WithBlockCtx(context.Background(),
+		protocol.BlockCtx{
+			BlockHeight: uint64(1),
+			Producer:    identityset.Address(27),
+			GasLimit:    gasLimit,
+		})
+	zctx = protocol.WithBlockchainCtx(
+		zctx,
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(zctx))
+
+	t.Run("normal call", func(t *testing.T) {
+		blk := makeBlock(t, 0, hash.ZeroHash256, hash.Hash256b([]byte{65, 99, 99, 111, 117, 110, 116, 99, 117, 114, 114, 101, 110, 116, 72, 101, 105, 103, 104, 116, 1, 0, 0, 0, 0, 0, 0, 0}))
+		require.NoError(sf.Validate(zctx, blk))
```

Comment:
if use nonce = 0, build block should fail?

Reply:
https://github.com/iotexproject/iotex-core/blob/4c354b099f0c61738af5e2fe930637673615ea62/state/factory/factory_test.go#L725-L728

I refer here, there is no error in this place, and during my test, it is ok to use 0 or 1.